app/forms/hyrax/forms/resource_form.rb

```diff
@@ -67,6 +68,8 @@ class << self
         #   monograph  = Monograph.new
         #   change_set = Hyrax::Forms::ResourceForm.for(monograph)
         def for(work)
+          "#{work.class}Form".constantize.new(work)
```

Comment:
i was slightly reluctant to re-introduce this name-inference convention, but decided it was a good-enough starting place to get this work surfaced.

thoughts on how to do this better are welcome.

Reply:
You could ask the `work.class` for its form (set as a class attribute)

Reply:
Which might help provide a waypoint when someone looks at the model.

Reply:
except i think forms don't (necessarily) have a 1:1 relationship with a work. they're triangulated with a controller.

Reply:
For the yaml configuration the field can be defined as primary. that configuration would be used for any form associated with the work?

Reply:
> For the yaml configuration the field can be defined as primary. that configuration would be used for any form associated with the work?

this is a good question.

in the current setup, we have a dependency chain:
  - your form class
  - `ResourceForm`
  - `FormFields`
  - `SimpleSchemaLoader`
  - yaml config

if you wanted multiple forms for a specific work, you could either have multiple schemas like: 

```ruby
MainForm.include Hyrax::FormFields(:main_schema)
SpecializedForm.include Hyrax::FormFields(:specialized_schema)
```

or you could have a single schema two separate form configurations, and a custom loader. something like:

```ruby
MainForm.include Hyrax::FormFields(:schema, definition_loader: MyFormLoader.new(form: :main)
SpecializedForm.include Hyrax::FormFields(:schema, definition_loader: MyFormLoader.new(form: :specialized)
```

in either case, you'd then need to select the appropriate form from the controller. at the moment, this `.for` method does that for `WorksControllerBehavior` controllers; so in some sense there's an assumption that `ResourceForm` is for use by those controllers. whether that's something to stick to, i'm unsure.

Reply:
👍 great; this might be good documentation to find a more permanent / visible home for. It's tempting to suggest that different forms could be specified in the configuration, but it seems like probably an uncommon use case, which this process will support in a straightforward way.